aggregate-snapshot/aggregate-snapshot-domain/src/main/java/uk/gov/justice/domain/snapshot/AggregateSnapshot.java

```diff
@@ -18,25 +19,29 @@
 
 public class AggregateSnapshot<T extends Aggregate> implements Serializable {
 
+    private static final long serialVersionUID = -6621681121087097921L;
+
     private final UUID streamId;
     private final Long positionInStream;
     private final String type;
     private final byte[] aggregateByteRepresentation;
+    private final ZonedDateTime createdAt;
 
     @SuppressWarnings("unchecked")
-    public AggregateSnapshot(final UUID streamId, final Long versionId, final T aggregate) {
-        this(streamId, versionId, (Class<T>) aggregate.getClass(), serialize(aggregate));
+    public AggregateSnapshot(final UUID streamId, final Long versionId, final T aggregate, final ZonedDateTime createdAt) {
+        this(streamId, versionId, (Class<T>) aggregate.getClass(), serialize(aggregate), createdAt);
     }
 
-    public AggregateSnapshot(final UUID streamId, final Long versionId, final Class<T> type, final byte[] aggregateByteRepresentation) {
-        this(streamId, versionId, type.getName(), aggregateByteRepresentation);
+    public AggregateSnapshot(final UUID streamId, final Long versionId, final Class<T> type, final byte[] aggregateByteRepresentation, final ZonedDateTime createdAt) {
+        this(streamId, versionId, type.getName(), aggregateByteRepresentation, createdAt);
     }
 
-    public AggregateSnapshot(final UUID streamId, final Long versionId, final String type, final byte[] aggregateByteRepresentation) {
+    public AggregateSnapshot(final UUID streamId, final Long versionId, final String type, final byte[] aggregateByteRepresentation, final ZonedDateTime createdAt) {
```

Comment:
If the aggregateByteRepresentation byte array and createdAt date are linked, should they be but into a new single Object, AggregateByteArray or something. Also, if it's not always been created, should that Object be Optional?

Reply:
fixed